hack/build-and-push.sh

```diff
@@ -431,6 +434,34 @@ attach_migration_file() {
     return 0
 }
 
+# Find previous bundle that has a migration and output its digest.
+find_previous_migration_bundle_digest() {
+    local -r task_name=${1:?Missing task name}
+    local -r task_version=${2:?Missing task version}
+    local -r repo=${TEST_REPO_NAME:-task-${task_name}}
+
+    local -r url_list_recent_tags="https://quay.io/api/v1/repository/${QUAY_NAMESPACE}/${repo}/tag/?onlyActiveTags=true&limit=100"
+    local -r expr_filter_tags=".tags[] | select(.name | test(\"^(${task_name}-)?${task_version}-[0-9a-f]+\$\")) | .manifest_digest"
+
+    local prev_bundle_digest has_migration
+
+    curl --fail -sL "$url_list_recent_tags" | jq -r "$expr_filter_tags" | \
+        while read -r manifest_digest; do
+            curl --fail -sL -o /tmp/manifest.json \
+                "https://quay.io/v2/${QUAY_NAMESPACE}/${repo}/manifests/${manifest_digest}"
+            prev_bundle_digest=$(jq -r '.annotations."dev.konflux-ci.task/previous-migration-bundle"' </tmp/manifest.json)
+            has_migration=$(jq -r '.annotations."dev.konflux-ci.task.migration"' </tmp/manifest.json)
```

Comment:
nitpick: maybe we should keep the annotation style consistent
```suggestion
            prev_bundle_digest=$(jq -r '.annotations."dev.konflux-ci.task.previous-migration-bundle"' </tmp/manifest.json)
            has_migration=$(jq -r '.annotations."dev.konflux-ci.task.migration"' </tmp/manifest.json)
```